resources/user.rb

```diff
@@ -245,15 +245,19 @@ def revokify_key(key)
 
   # Repair
   if incorrect_privs
+    privileges_to_set = new_resource.privileges.map { |key| clean_grant_name(key) }
```

Comment:
I've not tested my PR against a RDS database, but this one won't work to my mind. In this specific case my PR better handles it by working with `desired_privs` instead of `new_resource.privileges`.
see [here for more details](https://aws.amazon.com/premiumsupport/knowledge-center/duplicate-master-user-mysql/)